v0/destinations/heap/transform.js

```diff
@@ -16,6 +16,10 @@ function responseBuilderSimple(message, category, destination) {
   if (payload) {
     if (payload.properties) {
       payload.properties = flattenJson(payload.properties);
+      // remove duplicate key as it is being passed at root.
+      if (payload.properties.idempotencyKey) {
```

Comment:
```suggestion
      if (get(payload,"properties.idempotencyKey")) {
```

Reply:
Even better,
`payload?.properties?.idempotencyKey`

Reply:
IMO it is not required as we already have `payload.properties` check above.

Reply:
What are the expected outcomes of the `flattenJson` method?

Reply:
converts nested layers into only one layer of key/value pair.

Reply:
Can it return `undefined` or `null` as a result of some internal error?

Reply:
No